python/tbb/test.py

```diff
@@ -82,6 +82,16 @@ def work(mseconds):
         say("[%d] Work done (%fms)." % (get_tid(), mseconds*10))
         return res
 
+    # special flag to to be set by thread calling async work
+    spin_flag = True
+    def timeout_work(param):
```

Comment:
Please consider also add separate function that will take user functor as argument and will implement smth like:
```python
def check_timeout_exception(func):
    spin_flag = true # lock threads in timeout_work
    try:
        func()
    except:
        msg()
    else:
        error

    spin_flag = false # unlock threads from timeout_work
```